hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -3388,7 +3387,8 @@ private CacheStoreMode determineCacheStoreMode(Map<String, Object> settings) {
 	}
 
 	private void checkTransactionNeeded() {
-		if ( !isTransactionInProgress() ) {
+		if ( !getSessionFactory().getSessionFactoryOptions().isAllowOutOfTransactionUpdateOperations()
```

Comment:
I don't agree with this, which seems like premature optimization.  Yes we
would incur the overhead of the method calls, but these are simple getter
calls.

On Wed, Jun 29, 2016, 6:52 AM Sanne Grinovero <notifications@github.com>
wrote:

> In hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java
> <https://github.com/hibernate/hibernate-orm/pull/1459#discussion_r68932605>
> :
>
> > @@ -3388,7 +3387,8 @@ private CacheStoreMode determineCacheStoreMode(Map<String, Object> settings) {
> >  	}
> >
> >  	private void checkTransactionNeeded() {
> > -		if ( !isTransactionInProgress() ) {
> > +		if ( !getSessionFactory().getSessionFactoryOptions().isAllowOutOfTransactionUpdateOperations()
>
> I'm wondering about performance here, as this method is invoked frequently.
>
> Specifically, the block:
>
> getSessionFactory().getSessionFactoryOptions().isAllowOutOfTransactionUpdateOperations()
>
> Considering that this value will never change at runtime, what about
> making it a field of SessionImpl and initialize it during constructor?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hibernate/hibernate-orm/pull/1459/files/3a6add6c362a56509ff0a917412f85a87997300b..a7b5c9e7da94baf0393bd77c8db6316363d458a7#r68932605>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAOUE-R1eNZReL6FVGRr0zqCmE8gP7i_ks5qQlyJgaJpZM4JAEeF>
> .
>